packages/vuelidate/test/unit/specs/validation.spec.js

```diff
@@ -35,6 +35,25 @@ describe('useVuelidate', () => {
     shouldBePristineValidationObj(vm.v.number)
   })
 
+  it('accepts a function as validations', async () => {
+    const password = ref('a special password')
+    const confirmPassword = ref('')
+
+    const { vm } = createSimpleWrapper(() => ({
+      confirmPassword: {
+        sameAs: sameAs(password.value)
+      }
+    }), { confirmPassword })
```

Comment:
How is this different than this? The value is a `ref` and is already available, so you can pass it, right?

```suggestion
    const { vm } = createSimpleWrapper({
      confirmPassword: {
        sameAs: sameAs(password)
      }
    }, { confirmPassword })
```

Reply:
Yup good catch - I will update that. Forgot that `unwrap` would handle the `.value`.

Reply:
No, i also removed the function wrapping it. I still dont see the reason for it

Reply:
Ah I didn't see that when I first looked at it. That was the whole change that this PR made. If you want to separate your validations from the component file there is no way to do that with how the hook currently works. Making a function allows you to pass in whatever you need from the `setup()` function. Here is an example (but let me know if there is a way to accomplish the same thing with the current version and maybe this PR is not necessary, I may just be misunderstanding how hooks work in Vue :sweat_smile: ):

```js
// in my_component.vue
import { UserValidations } from './user-validations';
import useVuelidate from '@vuelidate/core';
export default defineComponent({
  setup(){
    const user = reactive({ email: null, password: null, passwordConfirmation: null });
    const $v = useVuelidate(UserValidations(user), user);
  }
});

// in user-validations.js
import { email, required, sameAs } from '@vuelidate/validators';

export const UserValidations = ({password}) => () => ({
  email: { email },
  password: { required },
  passwordConfirmation: {
    sameAs: sameAs(password, 'Password')
  }
});
```

Reply:
And actually typing this out I think it actually might work without return a function :facepalm:

Reply:
Maybe not...I just tested it in my project and it is not using the reactive value

Reply:
Check your validations. You cannot distructure a reactive data. You have to transform it to refs.

Reply:
My mistake! Still new at the composition api :sweat_smile: thanks for setting me straight though. I just tried it using `toRefs` and it worked! :tada:

Reply:
Yay! Glad to see it worked! Totally understandable the composition-API has it quirks.